stat/statmat_test.go

```diff
@@ -303,6 +306,235 @@ func TestMahalanobis(t *testing.T) {
 	}
 }
 
+func flatten(f [][]float64) (r, c int, d []float64) {
+	r = len(f)
+	if r == 0 {
+		panic("bad test: no row")
+	}
+	c = len(f[0])
+	d = make([]float64, 0, r*c)
+	for _, row := range f {
+		if len(row) != c {
+			panic("bad test: ragged input")
+		}
+		d = append(d, row...)
+	}
+	return r, c, d
+}
+
+func TestLassoRegression(t *testing.T) {
+	// y = 2 + 3*x0 + 4*x1
+	testData := map[string]struct {
+		x         *mat.Dense
+		y         *mat.Dense
+		model     LassoModel
+		err       error
+		tol       float64
+		intercept float64
+		coef      []float64
+	}{
+		"invalid lambda": {
```

Comment:
```suggestion
		"invalid_lambda": {
```
(similar throughout; it simplifies finding cases)